### PR TITLE
test: add OpTransaction conversion tests

### LINKED ISSUE #1894: Test revm-optimism
Need to sync to check if all is fine, or if we missed something. 

First step would be to write few dummy tests. It is reported here https://github.com/bluealloy/revm/issues/1839 that current main is not working as expected

### PR DESCRIPTION
**Pull request by hoank101**
Test conversion of OpTransaction

### CODE REVIEW

File: crates/optimism/src/transaction/abstraction.rs [hunk 1]
```diff
@@ -181,3 +181,84 @@ impl<T: Transaction> OpTxTrait for OpTransaction<T> {
...
+        // Verify common fields access
+        assert_eq!(op_tx.common_fields().gas_limit(), 0);
+        assert_eq!(op_tx.kind(), revm::primitives::TxKind::Call(Address::ZERO));
+        // Verify gas related calculations
+        assert_eq!(op_tx.effective_gas_price(U256::from(100)), U256::from(100));
+        assert_eq!(op_tx.max_fee(), 0);
+    }
+
+    #[test]
+    fn test_deposit_transaction_type_access_panics() {
```
Comment (rakita) [inline crates/optimism/src/transaction/abstraction.rs:245]: Would remove this test, it looks strange

File: crates/optimism/src/transaction/abstraction.rs [hunk 2]
```diff
@@ -181,3 +181,84 @@ impl<T: Transaction> OpTxTrait for OpTransaction<T> {
         }
     }
 }
+
+#[cfg(test)]
+#[cfg(feature = "std")]
```
Comment (rakita) [inline crates/optimism/src/transaction/abstraction.rs:186]:
```suggestion
```

File: crates/optimism/src/transaction/abstraction.rs [hunk 3]
```diff
@@ -181,3 +181,84 @@ impl<T: Transaction> OpTxTrait for OpTransaction<T> {
...
+            assert!(result.is_err());
+            if let Err(err) = result {
+                if let Some(message) = err.downcast_ref::<&str>() {
+                    assert_eq!(*message, $expected);
+                } else {
+                    panic!("Panic occurred but with unexpected message type");
+                }
+            }
+        };
+    }
```
Comment (rakita) [inline crates/optimism/src/transaction/abstraction.rs:207]:
```suggestion
```

File: crates/optimism/src/transaction/abstraction.rs [hunk 4]
```diff
@@ -181,3 +181,84 @@ impl<T: Transaction> OpTxTrait for OpTransaction<T> {
...
+            input: Default::default(),
+        };
+        let op_tx: OpTransaction<TxEnv> = OpTransaction::Deposit(deposit);
+        // Test each transaction type access method
+        assert_panic!("Not a legacy transaction", op_tx.legacy());
+        assert_panic!("Not eip2930 transaction", op_tx.eip2930());
+        assert_panic!("Not a eip1559 transaction", op_tx.eip1559());
+        assert_panic!("Not a eip4844 transaction", op_tx.eip4844());
+        assert_panic!("Not a eip7702 transaction", op_tx.eip7702());
+    }
```
Comment (rakita) [inline crates/optimism/src/transaction/abstraction.rs:263]:
```suggestion
```

**Review by rakita — APPROVED** [review]
Good test, lgtm. Lets remove the panic ones.

Will change Issue name and description as it test only small part of op

**Comment by hoank101** [comment]
> Good test, lgtm. Lets remove the panic ones.
> 
> Will change Issue name and description as it test only small part of op

thank you for feedback, I just updated the code

**Pull request merged**